v1/lib/v1/search_engine/river.rb

```diff
@@ -165,29 +187,75 @@ def self.verify_river_status(name=river_name)
+                 "#{last_seq} running on node '#{node}'"
         end
       end
 
+      ##
+      # Get the last sequence number from the River
+      #
+      # This method used to return nil if HTTParty.get failed. There were no
+      # unit tests, so we have to determine the correct behavior through
+      # analysis. `.recreate_river' above is now stricter about only
+      # _re_creating rivers, so maybe this function used to ignore errors
+      # because it was convenient to create a new river if one didn't exist.
+      #
+      # An exception will be raised if the River exists, but never started
+      # flowing because the source CouchDB database is still empty.  In that
+      # case, the HTTP GET request to the River endpoint will return HTTP
+      # status 404 and the response body will be a JSON object without
+      # the '_source' property. The remedy to this is to require a last
+      # sequence number when creating a River, which we now do, to eliminate
+      # ambiguity.  A last_seq property of "0" can be specified in that case.
+      #
+      # @return [Fixnum]       Sequence number
+      # @param  name [String]  Name of the River
+      # @raise  [RuntimeError] if the HTTP request to the River endpoint fails
+      # @see self.recreate_river
+      #
       def self.last_sequence(name=river_name)
         # name will be present but nil when called from a rake task
         name ||= river_name
-        response = HTTParty.get(endpoint(name) + '/_seq').parsed_response
-        seq = response['_source']['couchdb']['last_seq'] rescue nil
-
-        return nil if seq.nil?
+        response = HTTParty.get(endpoint(name) + '/_seq')
+        raise "Problem getting last sequence: #{JSON.parse(response.body)}" \
+          unless response.success?
+        seq = response.parsed_response['_source']['couchdb']['last_seq']
 
         # handle bigcouch's last_seq format if it looks like it
-        seq = JSON.parse(seq).first if seq !~ /^\d+$/
+        seq = JSON.parse(seq).first if seq !~ /^\d+$/  # but how not?  -mb
 
         seq.to_i
       end
 
+      ##
+      # Set the last sequence number for a River.
+      #
+      # The sequence number can be PUT for a River before the rest of it is
+      # created by a PUT to its _meta resource. Putting last_seq creates a data
+      # structure for the incomplete river. When you do this (before PUTing to
+      # `_meta') you'll see a property for the new river in `/_river/_mapping'.
+      #
+      # @param  seq_num [Fixnum or String] Sequence number /^\d+$/
+      # @return true
+      # @raise [RuntimeError]  if seq_num is improper type
+      # @raise [RuntimeError]  if HTTP transaction fails
+      #
+      def self.last_sequence!(seq_num, rname=river_name)
+        # rname will be present but nil when called from a rake task
+        rname ||= river_name
+        seq_num_s = seq_num.to_s
+        raise "Bad sequence number #{seq_num_s}" if seq_num_s !~ /^\d+$/
```

Comment:
I know about the array format, but I'm not sure I understand, because the only way `last_sequence!` is going to get `seq_num` is from the call in `.create_river`, where the value is either going to be an integer (which a calling function has got from `.last_sequence` ) or a string that's been entered by a person invoking the rake task (where the person should be in control of putting in a number). Do you know of another place where we might be getting some non-numeric value? Note that we're getting the last sequence number in our code from the ES river endpoint (`/_river/rivername/_seq`) and that parses out and returns the numeric value.  Example:
```
{
  "_index": "_river",
  "_type": "mbtest_river",
  "_id": "_seq",
  "_version": 1,
  "exists": true,
  "_source": {
    "couchdb": {
      "last_seq": "0"
    }
  }
}
```